Jenkinsfile

```diff
@@ -29,7 +28,11 @@ for(j = 0; j < jdks.size(); j++) {
     def jdk = jdks[j]
     builds["${buildType}-jdk${jdk}"] = {
         // see https://github.com/jenkins-infra/documentation/blob/master/ci.adoc#node-labels for information on what node types are available
-        node(buildType == 'Linux' ? (jdk == 8 ? 'maven' : 'maven-11') : buildType.toLowerCase()) {
+        String agentContainerLabel = jdk == 8 ? 'maven' : 'maven-11'
+        if (buildType == 'Windows') {
+            agentContainerLabel += '-windows'
```

Comment:
Do we really need four combinations? To save resources, and sometimes wall clock time, I think it would suffice to run, say, Linux builds on Java 11 and Windows builds on Java 8.

Reply:
Nah, I was planning to reduce this to two runs before marking this change as ready for review. But in the meantime I've been running the draft PR with 4 builds to shake out any flakiness.

BTW jenkinsci/jenkins#6029

Reply:
My bad. I marked this as draft.